examples/3_Advanced_Topics/3-2_Creating_an_XML_BoM.py

```diff
@@ -0,0 +1,247 @@
+)
+
+handle_pin_part = bom_types.Part(
+    part_number="H-PIN-12",
+    mass_per_unit_of_measure=bom_types.UnittedValue(46.5, "g/Part"),
+    materials=[bom_types.Material(mi_material_reference=steel_1015_reference, percentage=100.)]
+)
+
+handle_grub_screw_part = bom_types.Part(
+    part_number="SSF-M4-6-A2",
+    mass_per_unit_of_measure=bom_types.UnittedValue(1.3, "g/Part"),
+    materials=[bom_types.Material(mi_material_reference=hardened_stainless_reference, percentage=100.)]
+)
+
+handle_assembly = bom_types.Part(
+    part_number="H-S-BR-Dual",
+    components=[
+        add_part_to_assembly_with_count(handle_part_a, 1),
+        add_part_to_assembly_with_count(handle_part_b, 1),
+        add_part_to_assembly_with_count(handle_pin_part, 2),
+        add_part_to_assembly_with_count(washer_part, 4),
+        add_part_to_assembly_with_count(handle_grub_screw_part, 2)
+    ]
+)
+# -
+
+# The glass panel consists of a laminated glass door and a layer of PET solar control
+# film
+
+# +
+glass_panel = bom_types.Part(
+    part_number="321-51",
+    mass_per_unit_of_measure=bom_types.UnittedValue(19.6, "kg/m^2"),
+    materials=[bom_types.Material(mi_material_reference=laminated_glass_reference, percentage=100.)])
+
+solar_control_film = bom_types.Part(
+    part_number="7000001298",
+    mass_per_unit_of_measure=bom_types.UnittedValue(340., "g/m^2"),
+    materials=[bom_types.Material(mi_material_reference=pet_film_reference, percentage=100.)]
+)
+
+panel_assembly = bom_types.Part(
+    part_number="P-30-L",
+    components=[
+        add_part_to_assembly_with_quantity(glass_panel, 1.51, "m^2"),
+        add_part_to_assembly_with_quantity(solar_control_film, 1.51, "m^2"),
+    ]
+)
+# -
+
+# The whole door assembly is then a combination of two hinges, one handle assembly and one
+# door panel
+
+# +
+door_assembly = bom_types.Part(
+    part_number="24X6-30",
+    components=[
+        add_part_to_assembly_with_count(hinge_assembly, 2),
+        add_part_to_assembly_with_count(handle_assembly, 1),
+        add_part_to_assembly_with_count(panel_assembly, 1),
+    ]
+)
+# -
+
+# Generate a bom from the door assembly part
+
+# +
+from ansys.grantami.bomanalytics import BoMHandler
+
+door_assembly_bom = bom_types.BillOfMaterials(components=[door_assembly])
+bom_handler = BoMHandler()
+
+rendered_bom = bom_handler.dump_bom(door_assembly_bom)
+rendered_bom.splitlines()[0:10]
+# -
+
+# Now that you have created an XML BoM, run a compliance query to determine whether the BoM complies
+# with a specific legislation.
+# First, connect to Granta MI.
+
+# +
+from ansys.grantami.bomanalytics import Connection
+
+server_url = "http://my_grantami_server/mi_servicelayer"
+cxn = Connection(server_url).with_credentials("user_name", "password").connect()
+# -
+
+# The impacted substances BoM query behaves similar to other impacted substances queries. However,
```

Comment:
This probably needs to be updated, to reflect the fact that you're running a compliance query, not an impacted substances query.